widget/embedded/src/components/SwapDetails/SwapDetails.helpers.tsx

```diff
@@ -0,0 +1,215 @@
+  PendingSwapStep,
+} from '@rango-dev/queue-manager-rango-preset';
+import type {
+  PriceImpactWarningLevel,
+  Step,
+  StepDetailsProps,
+} from '@rango-dev/ui';
+import type { ConnectResult } from '@rango-dev/wallets-react';
+import type BigNumber from 'bignumber.js';
+
+import { i18n } from '@lingui/core';
+import { PendingSwapNetworkStatus } from '@rango-dev/queue-manager-rango-preset';
+import { Alert, Button, InfoIcon, LinkIcon, Tooltip } from '@rango-dev/ui';
+import React, { Fragment } from 'react';
+
+import { numberToString } from '../../utils/numbers';
+import { isNetworkStatusInWarningState } from '../../utils/swap';
+
+import { Alerts } from './SwapDetails.styles';
+
+export const LOW_PRICE_IMPACT = 3;
+export const HIGHT_PRICE_IMPACT = 10;
+export const RESET_INTERVAL = 2_000;
+export const MIN_DECIMALS = 0;
+export const AMOUNT_DECIMALS = 4;
+export const PERCENT_DECIMALS = 2;
+export const SECONDS = 60;
+
+export const getSteps = (
+  pendingSwap: PendingSwap,
+  switchNetwork: (() => Promise<ConnectResult>) | undefined,
+  showNetworkModal: PendingSwapNetworkStatus | null | undefined,
+  setNetworkModal: (network: PendingSwapNetworkStatus) => void,
+  message: ReturnType<typeof getSwapMessages>
+): Step[] => {
+  return pendingSwap.steps.map((step, index) => {
+    return {
+      from: {
+        token: { displayName: step.fromSymbol, image: step.fromLogo },
+        chain: {
+          displayName: step.fromBlockchain,
+          image: step.fromBlockchainLogo,
+        },
+        price: {
+          value:
+            index === 0
+              ? numberToString(
+                  pendingSwap.inputAmount,
+                  AMOUNT_DECIMALS,
+                  AMOUNT_DECIMALS
+                )
+              : numberToString(
+                  pendingSwap.steps[index - 1].outputAmount,
+                  AMOUNT_DECIMALS,
+                  AMOUNT_DECIMALS
+                ),
+        },
+      },
+      to: {
+        token: { displayName: step.toSymbol, image: step.toLogo },
+        chain: {
+          displayName: step.toBlockchain,
+          image: step.toBlockchainLogo,
+        },
+        price: {
+          value: step.outputAmount
+            ? numberToString(
+                step.outputAmount,
+                AMOUNT_DECIMALS,
+                AMOUNT_DECIMALS
+              )
+            : numberToString(
+                step.expectedOutputAmountHumanReadable,
+                AMOUNT_DECIMALS,
+                AMOUNT_DECIMALS
+              ),
+        },
+      },
+      swapper: { displayName: step.swapperId, image: step.swapperLogo },
+      alerts: (
+        <Alerts>
+          {step.explorerUrl?.map((explorerUrl, index, urls) => {
+            const key = index + explorerUrl.url;
+            return (
+              <Fragment key={key}>
+                <Alert
+                  type={
+                    (index === urls.length - 1 &&
+                      step.status === 'running' &&
```

Comment:
Please refactor this part.  (`type` value) It's not readable in this format.

Reply:
refactored.